src/main/java/org/robotframework/remoteserver/library/DynamicApiRemoteLibrary.java

```diff
@@ -41,7 +41,8 @@ protected DynamicApiRemoteLibrary(Object library, Method getKeywordNames, Method
         try {
             Object names = getKeywordNames.invoke(library, new Object[] {});
             if (names instanceof List) {
-                return (String[]) ((List<?>) names).toArray();
+                Object[] nameArray = ((List<?>) names).toArray();
+                return Arrays.copyOf(nameArray, nameArray.length, String[].class);
```

Comment:
You are really creating two copies of the array, wasting CPU time.

You should just use the generic-version of toArray: https://docs.oracle.com/javase/8/docs/api/java/util/List.html#toArray-T:A- 

something along the lines of

String [] nameArray = names.toArray(new String[names.size()]);

Reply:
toArray(<desired array type)[0]) seems to be faster: https://stackoverflow.com/questions/174093/toarraynew-myclass0-or-toarraynew-myclassmylist-size